include/crm_internal.h

```diff
@@ -79,7 +79,6 @@
 #define PCMK__ATTRD_CMD_UPDATE_DELAY    "update-delay"
 #define PCMK__ATTRD_CMD_QUERY           "query"
 #define PCMK__ATTRD_CMD_REFRESH         "refresh"
-#define PCMK__ATTRD_CMD_FLUSH           "flush"
```

Comment:
I'm actually inclined to leave this in for now. It doesn't matter currently because pacemaker-attrd ignores unknown IPC commands (in attrd_peer_message()), so the flush command will be accepted (and ignored) from older cluster peers. However that's not a very good design; nack'ing an unknown command would be a better approach, and at that time we'd need to start explicitly ignoring flush.

(Unfortunately nack'ing isn't as trivial as it sounds, we really need a comprehensive approach to ack/nack, discriminating by peer version so older peers don't get unexpected replies. Even better if we make it generically applicable to all daemon IPC -- which will be easier once we've converted the remaining APIs to the new pcmk_ipc_api_t interface.)

If the dangling constant bugs you, feel free to add the explicit ignore to attrd_peer_message() for future-proofing.

Reply:
> so the flush command will be accepted (and ignored) from older cluster peers

What older peers are we worried about? `ATTRD_OP_FLUSH` was already gone in 2.0.0. Do we have to handle peers running version 1? Genuine curiosity.

> feel free to add the explicit ignore to attrd_peer_message() for future-proofing.

Would this take the form of an empty `if` block?

Reply:
> > so the flush command will be accepted (and ignored) from older cluster peers
> 
> What older peers are we worried about? `ATTRD_OP_FLUSH` was already gone in 2.0.0. Do we have to handle peers running version 1? Genuine curiosity.

Yep, upstream we support mixed-version clusters as far back as 1.1.11.

> > feel free to add the explicit ignore to attrd_peer_message() for future-proofing.
> 
> Would this take the form of an empty `if` block?

Other than a comment :)

Reply:
> Yep, upstream we support mixed-version clusters as far back as 1.1.11.

It's worth noting that due to CRM feature set, that would be allowed only during a rolling upgrade, which ideally would be a relatively small time window. I wouldn't be surprised if things went south if you kept a cluster like that for an extended time.

In practice, distros added corosync 3 (which doesn't support rolling upgrades from corosync <=2) along with pacemaker 2 (which also dropped support for corosync <2), so the only users who would benefit would be those running pacemaker 1 on corosync 2 who build their own packages and stayed on corosync 2 for the pacemaker 2 rolling upgrade. Which is probably no one, but promises are promises :)